File: 3151 - Special Array I.py
class Solution:
    def isArraySpecial(self, nums: List[int]) -> bool:
        if len(nums) == 1:
            return True
        i = 1
        while i < len(nums) and nums[i - 1] % 2 != nums[i] % 2:
            i += 1
        return not i <= len(nums) - 1

File: 3151 - Special Array I.cpp
class Solution {
public:
    bool isArraySpecial(vector<int>& nums) {
        if (nums.size() == 1) return true;

        bool prev = nums[0] & 1;
        for (int i = 1; i < nums.size(); i++){
            bool curr = nums[i] & 1;
            if (prev == curr) return false;
            prev = curr;
        }
        return true;
    }
};